### DISCUSSION TITLE
Renovate creates PR/MR even if poetry.lock is failed

### DISCUSSION THREAD

**Discussion by Olegt0rr** [renovatebot/renovate#13623] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

Latest from https://gitlab.com/renovate-bot/renovate-runner/

### Please select which platform you are using if self-hosting.

gitlab.com

### If you're self-hosting Renovate, tell us what version of the platform you run.

https://gitlab.com

### Describe the bug

Latest renovate version `v31.28.3` skip updating of `poetry.lock` even if it's failed.

<img width="1211" alt="Screenshot 2022-01-18 at 15 42 43" src="https://user-images.githubusercontent.com/25399456/149939851-671b492b-c2b1-45a8-9827-58655578b9a5.png">

### Relevant debug logs
```
WARN: artifactErrors (repository=.... branch=renovate/dnspython-2.x)
       "artifactErrors": [
         {
           "lockFile": "poetry.lock",
           "stderr": 

SolverProblemError
Because no versions of pymongo match >3.12,<3.12.1 || >3.12.1,<3.12.2 || >3.12.2,<3.12.3 || >3.12.3,<3.13
   and pymongo (3.12.0) depends on dnspython (>=1.16.0,<1.17.0), pymongo (>=3.12,<3.12.1 || >3.12.1,<3.12.2 || >3.12.2,<3.12.3 || >3.12.3,<3.13) requires dnspython (>=1.16.0,<1.17.0).
  And because pymongo (3.12.1) depends on dnspython (>=1.16.0,<1.17.0), pymongo (>=3.12,<3.12.2 || >3.12.2,<3.12.3 || >3.12.3,<3.13) requires dnspython (>=1.16.0,<1.17.0).
  And because pymongo (3.12.2) depends on dnspython (>=1.16.0,<1.17.0)\n   and pymongo (3.12.3) depends on dnspython (>=1.16.0,<1.17.0), pymongo (>=3.12,<3.13) requires dnspython (>=1.16.0,<1.17.0).

So, because <project> depends on both dnspython (~2.1) and pymongo (~3.12), version solving failed.
```

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment, marked as answer]
Please describe in more detail what you mean.

Does Renovate create a MR and you think it shouldn't (in which case you are wrong because Renovate intentionally creates MRs in such cases so that you're aware of them instead of unaware).

Or does Renovate fail to create the MR? In which case we need to know more about your config and get debug logs.

**Reply by Olegt0rr** [reply]
@rarkins, 

1. Renovate detects new version of `dependency A`.
2. Renovate tries to generate new `poetry.lock`, but it fails, cause `dependency A` is incompatible with `dependency B`
3. Since the new version of `dependency A` will break my application - I think it's not a good idea to send me MR with it

**Reply by rarkins** [reply]
That way you can end up with no updates and be completely ignorant of it, especially every time there's a major update of a dependency. Renovate's design is to raise PRs so that you can choose to either ignore (close) it or add a commit on top to fix it before merging.

**Reply by rarkins** [reply]
Summary: feature operating as designed
